signalfx-java/src/main/java/com/signalfx/metrics/connection/HttpDataPointProtobufReceiverFactory.java

```diff
@@ -42,6 +49,11 @@ public HttpDataPointProtobufReceiverFactory setMaxRetries(int maxRetries) {
         return this;
     }
 
+    public HttpDataPointProtobufReceiverFactory setNonRetryableExceptions(List<Class<? extends IOException>> clazzes) {
+        this.nonRetryableExceptions = Collections.unmodifiableList(clazzes);
```

Comment:
`unmodifiableList()` doesn't copy the list, it just creates a thin immutable wrapper over it.
```suggestion
        this.nonRetryableExceptions = new ArrayList<>(clazzes);
```

Reply:
Ah ok, thanks for that insight, I might do `Collections.unmodifiableList(new ArrayList<>(clazzes))` instead then, this will create an unmodifiable view without any reference to the original ArrayList.